### PR TITLE
Fix not being able to filter nodes by searching exact label

### PR DESCRIPTION
**Pull request by rudream**
### Purpose

This PR addresses #268 

Users should be able to search through nodes by typing in the exact label/tag in the search bar. (eg. typing `arch: x86_64` in the search bar should return all nodes with `arch: x86_64` in its table column).

### Implementation

Instead of each node storing its labels as an array of objects (eg. `[{name: value}]`), it now stores them as an array of strings (eg. `["name: value"]`). This makes it possible for the labels to be directly compared against the string entered in the search bar, allowing for better and more predictable results when searching.

### Demo

demo-search

### CODE REVIEW

File: packages/teleport/src/services/nodes/makeNode.ts
Comment (kimlisa) [inline packages/teleport/src/services/nodes/makeNode.ts:21]: lets refactor this function to use the most recent way of processing responses (without lodash), similar to https://github.com/gravitational/webapps.e/blob/master/teleport/src/services/workflow/makeAccessRequest.ts#L8

Comment (rudream) [reply packages/teleport/src/services/nodes/makeNode.ts]: Done! :+1:

File: packages/teleport/src/services/nodes/makeNode.ts
Comment (kimlisa) [inline packages/teleport/src/services/nodes/makeNode.ts:22]: we can assign default values to destructured objects, so in line 20 we can do:
```ts
const {id, clusterId, hostname, addr, tunnel, tags = []} = json;
```

File: packages/teleport/src/services/nodes/makeNode.ts
Comment (kimlisa) [inline packages/teleport/src/services/nodes/makeNode.ts:23]: the json response won't contain a field named `tagTexts`, so checking if that is undefined is not needed, and since we supplied a default value for field `tags` (if `tags` came undefined), we can just do `tags.map` without worrying about undefined/null errors

Comment (rudream) [reply packages/teleport/src/services/nodes/makeNode.ts:23]: Yea, I wrote this before I found out the tests don't go through `makeNode`, so I though the `tagTexts` could be passed in

File: packages/teleport/src/Console/DocumentNodes/DocumentNodes.story.tsx
Comment (kimlisa) [inline packages/teleport/src/Console/DocumentNodes/DocumentNodes.story.tsx:21]: i don't think your prettier is working. i see double quotes here, but with prettify on save, should be converted to single quotes. If you followed the readme, i think its missing config for typescript files.

Could you update our [README](https://github.com/gravitational/webapps/blob/master/README.md#setup-prettier-on-vscode) with the following:

```
// Set the default
    "editor.formatOnSave": false,
    // absolute config path
    "prettier.configPath": ".prettierrc",
    // enable per-language
    "[html]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
    "[javascript]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
    "[javascriptreact]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "esbenp.prettier-vscode",
    },
    "[typescript]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
    "[typescriptreact]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "esbenp.prettier-vscode",
    },
    "[json]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "vscode.json-language-features"
    },
    "[jsonc]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "vscode.json-language-features"
    },
    "[markdown]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "esbenp.prettier-vscode",
    },
    "editor.tabSize": 2,
```

**Comment by klizhentas** [comment]
@rudream congrats on the first PR!  🚀

File: packages/teleport/src/services/nodes/makeNode.ts
Comment (alex-kovoy) [inline packages/teleport/src/services/nodes/makeNode.ts:28]: i would keep the `tag` name as it is shorter. 

```ts
const tags = json.tags || [];
return {
   ...json,
   tags: tags.map(tag => `${tag.name}: ${tag.value}`),
}
```

Comment (rudream) [reply packages/teleport/src/services/nodes/makeNode.ts]: Good idea. Done! :+1:

**Review by kimlisa — APPROVED** [review]
LGTM, but please wait for @alex-kovoy's approval before merge

btw, i found your gif very helpful :)

**Review by alex-kovoy — APPROVED** [review]

**Pull request merged**
